src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs

```diff
@@ -224,18 +224,60 @@ public static void Main(string[] args)
         [ConditionalFact(typeof(WindowsOnly))]
         public void NestedIfStatements()
         {
-            int nestingLevel = (ExecutionConditionUtil.Architecture, ExecutionConditionUtil.Configuration) switch
+            (int nestingLevelLower, int nestingLevelHigher) = (ExecutionConditionUtil.Architecture, ExecutionConditionUtil.Configuration) switch
```

Comment:
The tests need three different variations:

1. Limit without using threads: this verifies that we don't regress our current limits for executing on a single stack
1. Limit with using threads: essentially 10X of the (1) above. That should pass because we have *unlimited* stack space now 
1. Limit without using threads that crashes: the limit used in (2) should be tested in a context where we can't use more threads and verify that it fails with a diagnostic / exception. That verifies our limits are actually testing the code we wan to test.

Reply:
I'm concerned whether (3) or (2) would be reliable/predictable across systems and environment (memory pressure from other processes). What do you think?